### ISSUE TITLE
Can StanJulia join the party?

### ISSUE THREAD

**Issue by goedman** [TuringLang/MCMCChain.jl#10]
Hi Martin,

Just saw your email to Brian and we already had been thinking to contact the Turing team on this topic.

Right now we're kind of redoing the original Stan.jl by separating it into CmdStan.jl + a series of additional packages such as StanDataFrames.jl, StanMamba.jl and we've discussed the idea for common (Plots based) plotting, diagnostics and summary stats in Julia. StanJulia is also aiming to include storage for larger models and some other extensions such as StanLOO and StanARM.

All of the Stan related packages are part of the Github organization [StanJulia](https://github.com/StanJulia).

CmdStan.jl is basically the first more or less complete Julia 1.0 package in StanJulia. I would like to switch to Chain.jl (the interface could be something like StanChain.jl to create a TuringLang/Chains object). Applications would include a `using CmdStan, StanChain, Chain` statement to have access to all Chain capabilities.

To maintain backwards compatibility the next version of Stan.jl (v4.0) will wrap CmdStan.jl + StanMamba.jl to be feature compatible with pre-Julia 1.0 versions.

Would this be an ok approach for us to take?

Best,
Rob

Cc: @yebai @xukai92 @trappmartin @bparbhu

**Comment by yebai** [comment]
Hi @goedman, There is a lot in common between `Stan`, `StanJulia` and `Turing`. We'll be very happy with `StanJulia` to depend on `Chain.jl`! 

BTW, some users of `Turing` is also suggesting a high-level interface similar to `StanARM` (#460). So we are looking closely at `StanARM`. I'm enthusiastic about collaborating on using `Turing` as a backend for `StanARM`.

**Comment by goedman** [comment]
Thank you! I'll create the interface package (StanChain.jl) and start testing with the existing Stan examples.

Although these interface packages are adding very limited functionality they do provide insulation during transitions.

It would be really cool to have a good Julia implementation along the lines of the ARM book. The package should probably not be named *Stan*ARM but something more sampling engine neutral and more descriptive of its content (so well captured in the long title of the book).

**Comment by goedman** [comment]
Just a suggestion, I like the new name MCMCChain.jl but wonder if MCMCChains.jl would be more accurate.

**Comment by trappmartin** [comment]
Good point!

**Comment by DominiqueMakowski** [comment]
Although isn't the new name a bit redundant (impliying Makov *Chain* Monte Carlo *Chain(s)*) 😄 ?

**Comment by trappmartin** [comment]
Thanks for the comments, I'll rethink the name one last time.

**Comment by goedman** [comment]
Of course you are right. On the other hand,  ```MCMC``` is pretty established these days.

In the meantime, as a quick update, I have completed a first version of StanMCMCChains.jl. For now I have kept the plural in the name as that is 'suggested' for an object factory. It seems to work and pass Travis if I explicitly clone MCMCChain in .travis.yml. That of course will change once registered.

The GR (GKS) part in the bernoulli test will fail on Travis but works fine on my local (OSX) system.

On my system I have also tested with a forked version of MCMCChain.jl and added a Project.toml file to test in the Pkg3 setting. My experience has been that Pkg3 somehow seems to make it easier to get merged into METADATA (although lots of changes and improvements are still ongoing - just my 2 cents).
